packages/apollo-client/src/core/QueryManager.ts

```
@@ -1085,27 +1097,30 @@ export class QueryManager {
             this.broadcastQueries();
           }
 
-          // XXX remove to support errors in the store via policy
-          if (result.errors) {
+          if (result.errors && errorPolicy === 'none') {
             reject(
               new ApolloError({
                 graphQLErrors: result.errors,
               }),
             );
             return;
+          } else if (errorPolicy === 'all') {
+            errorsFromStore = result.errors;
           }
 
           if (fetchMoreForQueryId) {
             // We don't write fetchMore results to the store because this would overwrite
             // the original result in case an @connection directive is used.
             resultFromStore = result.data;
           } else {
-            // ensure result is combined with data already in store
-            resultFromStore = this.dataStore.getCache().read({
-              variables,
-              query: document,
-              optimistic: false,
-            });
+            try {
+              // ensure result is combined with data already in store
+              resultFromStore = this.dataStore.getCache().read({
+                variables,
+                query: document,
+                optimistic: false,
+              });
+            } catch (e) {}
```

Would be good to have a comment indicating why it is okay to ignore this error.